### PR TITLE
Updated theme to work via css variables

### PR DESCRIPTION
**Pull request by kozlovzxc-at-lido**
## Description

Also fixes UI-920

- moved `ContentTheme` & `ThemeToggler` components inside theme folder since they are directly related to theme functionality
- There is no `ThemeProvider`, which accepts any theme anymore
- Moved themes to singe file, it's easier to work with them, especially to set up custom selectors

### Content Theme

image
image

### Theme Providers

image
image

### Theme Toggler

image
image

Cookie is also supported
image

Manually changed to light
image

### System theme also supported

image

### CODE REVIEW

**Review by karinamaulitova — APPROVED** [review]

File: scripts/verify-allowed-css-variables.mjs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const getDefinedVariables = async () => {
+  const content = await fs.readFile('./packages/theme/theme.css', 'utf-8')
```
Comment (Jabher) [inline scripts/verify-allowed-css-variables.mjs:2]: leave annotation in theme.css that it is used here and should not be moved or renamed

File: packages/theme/utils/document-head-contents.tsx [hunk 2]
```diff
@@ -60,3 +62,25 @@ export const ScriptThemeValue = () => (
...
+
+export const Fonts = (): JSX.Element => <style>{CSS_FONTS}</style>
+
+export const LidoUIHead: FC<{
+  jsStyleScript?: boolean
+  cssFont?: boolean
+}> = ({ jsStyleScript = true, cssFont = false }) => (
+  <>
+    {jsStyleScript ? <ScriptThemeValue /> : null}
+    {cssFont ? <Fonts /> : null}
```
[the remark this answers is not in the document]
Comment (kozlovzxc-at-lido) [reply packages/theme/utils/document-head-contents.tsx:84]: It was false by default https://github.com/lidofinance/ui/pull/433/files#diff-57b896d8453f184a2877fce79694edbf146ff9a5dd4af8f06e0995f3dcfdb13eL16

File: packages/theme/theme-legacy.css [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  --lido-color-accentContrastSecondary: rgb(255 255 255 / 60%);
+  --lido-color-accentDarken: rgb(0 0 0 / 20%);
+  --lido-color-accentControlBg: rgb(39 39 46 / 56%);
+  --lido-color-accentBorder: rgb(255 255 255 / 12%);
+  --lido-color-accentBorderHover: rgb(255 255 255 / 24%);
+  --lido-color-error: #e14d4d;
+  --lido-rgb-error: 225, 77, 77;
+  --lido-color-errorHover: #d44c4d;
+  --lido-rgb-errorHover: 212, 76, 77;
+  --lido-color-errorContrast: #fff;
```
[the remark this answers is not in the document]
Comment (kozlovzxc-at-lido) [reply packages/theme/theme-legacy.css:190]: sure, will add

File: scripts/verify-allowed-css-variables.mjs (hunk 1 above)
Comment (Jabher) [reply scripts/verify-allowed-css-variables.mjs:2]: also, please add comment for this file - what is it doing?

File: scripts/verify-allowed-css-variables.mjs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    process.exit(0)
+  }
+
+  for (let weirdVar of usedButNotDefined) {
+    console.error(`var(${weirdVar}) used but not defined`)
+  }
+  process.exit(1)
+}
+
+void main()
```
Comment (Jabher) [inline scripts/verify-allowed-css-variables.mjs:44]: top-level await should be already supported, no need to wrap

**Review by Jabher — APPROVED** [review]

**Pull request merged**